src/components/Admin/SortIcon.jsx

```diff
@@ -5,21 +5,31 @@ import { TbTriangleFilled, TbTriangleInvertedFilled } from "react-icons/tb";
 
 const SortIcon = ({
   name,
-  sorts,
-  setSorts,
-  setfilteredObjects,
+  headers,
+  setHeaders,
+  setFilteredObjects,
   objects,
-  reset,
+  setCurrentSort,
+  currentSort,
 }) => {
   const handleClick = (state) => {
-    setSorts({ ...reset, [name]: state });
+    if (currentSort)
+      setHeaders({
+        ...headers,
+        [currentSort]: { ...headers[name], sort: "off" },
+      });
+    setHeaders({
+      ...headers,
+      [name]: { ...headers[name], sort: state },
+    });
 
-    setfilteredObjects(
+    setCurrentSort(name);
+    setFilteredObjects(
       objects.sort((a, b) => {
         if (state === "up") {
-          return a[name] > b[name] ? -1 : 1;
+          return a[name.toLowerCase()] > b[name.toLowerCase()] ? -1 : 1;
```

Comment:
we shouldn't have to use toLowerCase for anything handled internally, means we have capitalization errors we need to fix